test/lib/util/merge_test.js

```diff
@@ -211,7 +256,7 @@ describe('util: merge', function() {
             [ { foo: 'foo' }, null ],
             [ 1, NaN, 2, undefined, 0.4, true, 'foo', Infinity ]
         ].forEach(function(args) {
-            var expectedResult = args[args.length - 1];
+            var expectedResult = args[0];
```

Comment:
No, the merge should replace values from right to left.

    merge('a', 'b', 'c'); // should return 'c'